tests/typed_ir.tests

```diff
@@ -69,11 +69,21 @@ tests/typed_ir/T697b.links
 filemode : true
 stdout : () : ()
 
-Wild effect compatibility #3 (#873)
+Wild effect compatibility #3 (#873, 1)
 tests/typed_ir/T873.links
 filemode : true
-exit : 1
-stderr : @.*IR Type Error: Incompatible effects.*
+exit : 0
+stdout : () : ()
+
+Wild effect compatibility #4 (#873, 2)
+fun () { escape e in { (fun (esc) { esc(()) })(e) } }
+exit : 0
+stdout : fun : forall a::Row.() ~a~> ()
+
+Wild effect compatibility #5 (#873, 3)
+sig foo : () {a}~> () fun foo () { escape e in { (fun (esc) { esc(()) })(e) }}
```

Comment:
What's the point of `a` (`= a : () {}-> ()`) here?

Reply:
I'm not sure if there's a deeper reason, I've taken the test cases 2 and 3 in verbatim as things @jamescheney experimented with in issue #873.

Reply:
I have my own agenda here, I want to disallow `{name}` as syntactic sugar for `{name:()}` in order to fix #975 -- I essentially want to use the syntax `{name}` to mean presence variable in all types (it already does in rows). We will probably discuss this at the next meeting.

Reply:
In the examples I wrote,  I think I meant for `a` to be a (row) type variable, anyway.  So this test case could be changed to something like 

```
 sig foo : () { |e}~> () fun foo () { escape e in { (fun (esc) { esc(()) })(e) }}
```

(assuming that in fact works too)

Reply:
done, the test works with this modification